### PR TITLE
Implement target branch detection for the documentation repository

### LINKED ISSUE #6: Detect incorrect target branch in documentation
It happens a lot that PR's in the documentation repository is against the incorrect target branch. In those cases, we explain the user the difference and ask the user to change the target and rebase if needed.

The sooner we get to those cases, the most likely the user will "fix" this. So automating that, might be a good idea.

- When a PR on the repo `home-assistant/home-assistant.io` is edited or create and (now) contains a reference to a parent PR in any of the `home-assistant/*` repositories, *ensure* it is targeted against the `next` branch, if not:
    * Add the `needs-rebase` label
    * Add the `in-progress` label
    * Assign the PR author
    * Leave the user a message:
        > It seems that this PR is targeted against an incorrect branch since it has a parent PR on one of our codebases. Documentation that needs to be updated for an upcoming release should target the `next` branch. Please change the target branch of this PR to `next` and rebase if needed.

- When a PR on the repo `home-assistant/home-assistant.io` is edited or create and (now) *does not* a reference to a parent PR in any of the `home-assistant/*` repositories, *ensure* it is targeted against the `current` branch, if not:
    * Add the `needs-rebase` label
    * Add the `in-progress` label
    * Assign the PR author
    * Leave the user a message:
        > It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release need should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed.

### PR DESCRIPTION
**Pull request by ludeeus**
Closes #6

### CODE REVIEW

File: src/plugins/DocsTargetBranch/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Docs Target Branch
+
+Repositories: home-assistant.io
+
+The goal fo this bot is to keep track if documentation PRs have the corerct target branch.
```
Comment (MartinHjelmare) [inline src/plugins/DocsTargetBranch/README.md]: The goal of this bot is to determine if the documentation PRs have the correct target branch.

File: src/plugins/DocsTargetBranch/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+## Implemented Rules
+
+<!-- https://github.com/home-assistant/probot-home-assistant/issues/6 -->
+
+- When a PR on the repo `home-assistant/home-assistant.io` is edited or create and (now) contains a reference to a parent PR in any of the `home-assistant/*` repositories, ensure it is targeted against the `next` branch, if not:
```
Comment (MartinHjelmare) [inline src/plugins/DocsTargetBranch/README.md]: is edited or created

File: src/plugins/DocsTargetBranch/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - Add the `needs-rebase` label
+  - Add the `in-progress` label
+  - Assign the PR author
+  - Leave the user a message:
+    > It seems that this PR is targeted against an incorrect branch since it has a parent PR on one of our codebases. Documentation that needs to be updated for an upcoming release should target the `next` branch. Please change the target branch of this PR to `next` and rebase if needed.
+- When a PR on the repo `home-assistant/home-assistant.io` is edited or create and (now) does not a reference to a parent PR in any of the `home-assistant/*` repositories, ensure it is targeted against the `current` branch, if not:
```
Comment (MartinHjelmare) [inline src/plugins/DocsTargetBranch/README.md]: is edited or created

File: src/plugins/DocsTargetBranch/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+  - Add the needs-rebase label
+  - Add the in-progress label
+  - Assign the PR author
+  - Leave the user a message:
+    > It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release need should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed.
```
Comment (MartinHjelmare) [inline src/plugins/DocsTargetBranch/README.md]: Documentation updates which apply to our current stable release should target the `current` branch.

File: src/plugins/DocsTargetBranch/README.md (hunk 3 above)
Comment (MartinHjelmare) [reply src/plugins/DocsTargetBranch/README.md]: and (now) does not have a reference to a parent PR

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+import {
+  extractIssuesOrPullRequestMarkdownLinks,
+  extractPullRequestURLLinks,
+} from "../../util/text_parser";
+
+const NAME = "DocsTargetBranch";
+const SKIP_REPOS = ["brands"];
+
+export const bodyShouldTargetCurrent: string =
+  "It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release need should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed.";
```
Comment (MartinHjelmare) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts]: See language above.

**Review by MartinHjelmare — CHANGES_REQUESTED** [review]
I've only reviewed the text language not the code.

**Review by MartinHjelmare — COMMENTED** [review]
Thanks.

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+import {
+  extractIssuesOrPullRequestMarkdownLinks,
+  extractPullRequestURLLinks,
+} from "../../util/text_parser";
+
+const NAME = "DocsTargetBranch";
+const SKIP_REPOS = ["brands"];
+
+export const bodyShouldTargetCurrent: string =
+  "It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed.";
```
Comment (balloob) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts]:
```suggestion
  "It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed. If this is documentation for a new feature, please add a link to that PR in your description.";
```

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+import { filterEventByRepo } from "../../util/filter_event_repo";
+import { filterEventNoBot } from "../../util/filter_event_no_bot";
+import { REPO_HOME_ASSISTANT_IO, ORG_HASS } from "../../const";
+import {
+  extractIssuesOrPullRequestMarkdownLinks,
+  extractPullRequestURLLinks,
+} from "../../util/text_parser";
+
+const NAME = "DocsTargetBranch";
+const SKIP_REPOS = ["brands"];
```
Comment (balloob) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts]: Let's also skip the dev docs. 

btw let's use constants for these, in case we rename repos :)

Comment (balloob) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts]: https://github.com/home-assistant/probot-home-assistant/blob/master/src/const.ts#L2

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+
+export const bodyShouldTargetCurrent: string =
+  "It seems that this PR is targeted against an incorrect branch. Documentation updates which apply to our current stable release should target the `current` branch. Please change the target branch of this PR to `current` and rebase if needed.";
+export const bodyShouldTargetNext: string =
+  "It seems that this PR is targeted against an incorrect branch since it has a parent PR on one of our codebases. Documentation that needs to be updated for an upcoming release should target the `next` branch. Please change the target branch of this PR to `next` and rebase if needed.";
+
+export const initDocsTargetBranch = (app: Application) => {
+  app.on(
+    "pull_request.opened",
```
Comment (balloob) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts]: You only filter on when pull requests are opened, not on when they are updated ?

File: src/plugins/DocsTargetBranch/docs_target_branch.ts (hunk 7 above)
Comment (ludeeus) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts]: Moved and changed 👍

File: src/plugins/DocsTargetBranch/docs_target_branch.ts (hunk 8 above)
Comment (ludeeus) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts]: The way I read the issue I only thought about opened events, added `pull_request.edited` as well now 👍

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+  }
+
+  if (target !== "next") {
+    await wrongTargetBranchDetected(context, "next");
+  }
+};
+
+const wrongTargetBranchDetected = async (
+  context: PRContext,
+  correctTargetBranch: "current" | "next"
```
Comment (balloob) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts:60]: bitwise operator?

Comment (ludeeus) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts:60]: Not sure what you are asking here?

Comment (balloob) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts:60]: oh wait it's a type 😆 Good morning Paulus…

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+) => {
+  const labels = ["needs-rebase", "in-progress"];
+  const author = context.payload.sender.login;
+  const promises: Promise<unknown>[] = [];
+  const body: string = `${
```
Comment (balloob) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts]: no need to wrap in a "`${"

File: src/plugins/DocsTargetBranch/docs_target_branch.ts [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+    context.github.issues.addLabels(
+      // Bug in Probot: https://github.com/probot/probot/issues/917
+      // @ts-ignore
+      context.issue({
+        labels,
+      })
+    )
+  );
+
+  context.log(NAME, `Addin ${author} as assignee`);
```
Comment (balloob) [inline src/plugins/DocsTargetBranch/docs_target_branch.ts]:
```suggestion
  context.log(NAME, `Adding ${author} as assignee`);
```

Comment (balloob) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts]: btw assignee only works if they are part of the org, but that's fine.

Comment (ludeeus) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts]: True, I'll update the README with that.

File: src/plugins/DocsTargetBranch/docs_target_branch.ts (hunk 10 above)
Comment (ludeeus) [reply src/plugins/DocsTargetBranch/docs_target_branch.ts]: Yeah... I had some issues with the testing of the comment, tried multiple things but forgot to reset this one.

**Pull request merged**